src/ShardingConfigurationSource.php

```diff
@@ -5,25 +5,31 @@
 use PHPLRPM\Cluster\Filters\ShardingConfigurationFilter;
 use PHPLRPM\Cluster\Providers\ClusterConfigurationProvider;
 use PHPLRPM\ConfigurationSource;
+use RuntimeException;
 
-class ShardingConfigurationSource implements ConfigurationSource
+abstract class ShardingConfigurationSource implements ConfigurationSource
 {
-    private $confSource;
-    private $clusterConfProvider;
-    /**
-     * @var ShardingConfigurationFilter
-     */
-    private $configurationFilter;
+    /** @var ConfigurationSource */
+    protected $confSource;
 
-    public function __construct(ConfigurationSource $confSource, ClusterConfigurationProvider $clusterConfProvider, ShardingConfigurationFilter $configurationFilter)
-    {
+    /** @var ClusterConfigurationProvider */
+    protected $clusterConfProvider;
+
+    /** @var ShardingConfigurationFilter */
+    protected $configurationFilter;
+
+    abstract public function __construct();
+    /*
+     {
```

Comment:
Let's remove commented out code... We might include an example, maybe even better in the README.